src/main/java/org/myrobotlab/service/interfaces/TextListener.java

```diff
@@ -6,6 +6,24 @@ public interface TextListener extends NameProvider {
 
   public void onText(String text) throws Exception;
 
-  public void attachTextPublisher(TextPublisher service) throws Exception;
+  /**
+   * Attach a text listener
+   * 
+   * @param service
+   */
+  default public void attachTextPublisher(TextPublisher service) {

```

Comment:
This is backwards, we agreed that publishers would attach listeners.  Please clean this up to make it consistent with the design pattern for UtterancePublisher / UtteranceListener.

Reply:
No, I disagree  ...  if appropriate pub/sub is followed bi-directional attachment is trivial.
This interface implements the correct pub/sub call 
send(name, "attachTextListener", getName());
It works well without the burden of the user having to know which is the publisher and which is the user.
I would prefer you update your utterance listeners to do the same.

Reply:
sorry.. we agreed directionality is important and we should have a convention.  this has ramifications in the config generation...  publishers are supposed to attach listeners in their configs.

Reply:
configs are different than calling methods
publisher.attach(listener) works just as well as listener.attach(publisher)
publisher is the only one saving the attachment as config
all listener does is send a message through the default interface to add itself to the publisher
there is no ambiguity, or confusion - its simple and all implemented as pub/sub message sending